packages/react/src/scroll-area/viewport/ScrollAreaViewport.tsx

```diff
@@ -225,7 +227,16 @@ export const ScrollAreaViewport = React.forwardRef(function ScrollAreaViewport(
         });
       }
 
-      programmaticScrollRef.current = true;
+      // Debounce the restoration of the programmatic flag so that it only
+      // flips back to `true` once scrolling has come to a rest. This ensures
+      // that momentum scrolling (where no further user-interaction events fire)
+      // is still treated as user-driven.
+      scrollEndTimeout.clear();
+      // 100 ms without scroll events ≈ scroll end
+      // https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollend_event
+      scrollEndTimeout.start(100, () => {
+        programmaticScrollRef.current = true;
+      });
```

Comment:
`.clear()` is redundant. A `Timeout` instance can only contain one timeout at once, it clears the old one if a new one is started.